packages/nimble-components/src/table/specs/table-column-specs/table-column-anchor-hld.md

```diff
@@ -0,0 +1,121 @@
+    ...
+</nimble-table>
+```
+Note that this implies that these configuration parameters must be the same for each link in the column:
+- `target`
+- `queryParams`
+- `queryParamsHandling`
+- `fragment`
+- `state`
+- `relativeTo`
+- `preserveFragment`
+- `skipLocationChange`
+- `replaceUrl`
+
+Normally with RouterLink directives, you assign the url to `routerLink`/`nimbleRouterLink`, but in this case, the url is coming from the table data. `nimbleRouterLink` just has to be present for the directive to be applied.
+
+There is a problem with puting the directive on `nimble-table-column-anchor`, though. `RouterLinkWithHref` works by intercepting a click event that is bubbled up from a decendant link, but the `nimble-table-column-anchor` is not an ancestor of the anchor/table cell. The solution is to introduce a second directive that is on `nimble-table`, which _is_ an ancestor of the anchor/table cell. This directive will be responsible for intercepting click events from anchor elements, then "forwarding" the event to the right column directive.
+
+The first part of this task is to detect which click events have come from a `nimble-anchor`. We can't just look at the event's `target`, because that is changed to the host control upon crossing a shadow DOM boundary. We can, however, look at the `composedPath()` of the event, which will include every element that was along the bubble-up path. If we find a `nimble-anchor` among them (by using `instanceof`), we know we should handle the event.
+
+To handle the event, we must find the associated anchor column directive. We have the `nimble-anchor` that the event came from, and by calling `anchor.getRootNode().host`, we can get the `nimble-table-cell`. Currently, there is no way to get from a given table cell to its associated column. One way to solve this is to include the associated `columnId` in the cell state, i.e. `cell.cellState.columnId`.
+```ts
+export interface TableCellState<...> {
+    cellRecord: TCellRecord;
+    columnConfig: TColumnConfig;
+    columnId: string | undefined;
+}
+```
+`columnId` is at the top level rather than part of the `columnConfig`, because it applies for all column types, and it is not for configuration.
+Adding `columnId` to the cell state would be a change to `nimble-components` just to support `nimble-angular`, but it seems generic enough to be a reasonable API change.
+
+One problem is that `columnId` is `undefined` unless a client sets it. We would also need to start auto-generating unique `columnId` values if the client hasn't provided one (i.e. in `connectedCallback`). An alternative would be to document that users must provide column ids when wishing to integrate with the Angular router, but this seems like an arbitrary requirement that would be better to avoid. Another option considered was to reuse the existing `internalUniqueId` that each column defines, but this is used for the action menu slot, and overloading its use as a general-purpose ID seems wrong.
```

Comment:
>this seems like an arbitrary requirement that would be better to avoid

I'd personally be ok with requiring clilents to provide an ID for this case until someone pushes back. It's not too onerous of a requirement and it could save us work and complexity.

Reply:
The only work and complexity it would save us is auto-generating values, and that is pretty straightforward. But let's see if others chime in with opinions.

Reply:
Changed approach to use `internalUniqueId` instead of `columnId`, so this is moot.